opcodes.c

```diff
@@ -42,18 +42,18 @@
 #define RX_ADDRESS ((OPCODE & 7) + 8 * ((PSW & (PSWMASK_RS0|PSWMASK_RS1))>>PSW_RS0))
 #define CARRY ((PSW & PSWMASK_C) >> PSW_C)
 
-static int read_mem(struct em8051 *aCPU, int aAddress)
+static int read_mem(struct em8051 *aCPU, int address)
 {
-    if (aAddress > 0x7f)
+    if (address > 0x7f)
     {
-        if (aCPU->sfrread)
```

Comment:
Instead of -0x80'ing everywhere, just decrement aAddress here.

Reply:
When I looked at the generated code, it automatically removed the common sub-expression, and it felt more consistent that way.

But done in 471ebb4b395ca03316ed8dbb44b8ecc6c8b47e5a.